### PR TITLE
fix documentation warnings & 404 deadlink

### PR DESCRIPTION
**Pull request by leckie-chn**
```
/mnt/c/Users/yann/Desktop/code/nni/docs/en_US/community_sharings.rst:7: WARNING: toctree contains reference to nonexisting document 'CommunitySharings/NasComparison'
/mnt/c/Users/yann/Desktop/code/nni/docs/en_US/community_sharings.rst:7: WARNING: toctree contains reference to nonexisting document 'CommunitySharings/HpoComparison'
WARNING: autodoc: failed to import class 'smac_tuner.SMACTuner' from module 'nni.smac_tuner'; the following exception was raised:                   No module named 'ConfigSpaceNNI'
/mnt/c/Users/yann/Desktop/code/nni/src/sdk/pynni/nni/bohb_advisor/bohb_advisor.py:docstring of nni.bohb_advisor.bohb_advisor.BOHB:11: WARNING: Unexpected indentation.
/mnt/c/Users/yann/Desktop/code/nni/src/sdk/pynni/nni/bohb_advisor/bohb_advisor.py:docstring of nni.bohb_advisor.bohb_advisor.BOHB:13: WARNING: Block quote ends without a blank line; unexpected unindent.
/mnt/c/Users/yann/Desktop/code/nni/src/sdk/pynni/nni/bohb_advisor/bohb_advisor.py:docstring of nni.bohb_advisor.bohb_advisor.BOHB:25: WARNING: Unexpected indentation.
/mnt/c/Users/yann/Desktop/code/nni/src/sdk/pynni/nni/bohb_advisor/bohb_advisor.py:docstring of nni.bohb_advisor.bohb_advisor.BOHB:26: WARNING: Block quote ends without a blank line; unexpected unindent.

checking consistency... /mnt/c/Users/yann/Desktop/code/nni/docs/en_US/CommunitySharings/HpoComparision.md: WARNING: document isn't included in any toctree
/mnt/c/Users/yann/Desktop/code/nni/docs/en_US/CommunitySharings/NasComparision.md: WARNING: document isn't included in any toctree                  /mnt/c/Users/yann/Desktop/code/nni/docs/en_US/GPTuner.md: WARNING: document isn't included in any toctree                                           /mnt/c/Users/yann/Desktop/code/nni/docs/en_US/NniOnWindows.md: WARNING: document isn't included in any toctree                                      /mnt/c/Users/yann/Desktop/code/nni/docs/en_US/builtin_assessor.rst: WARNING: document isn't included in any toctree                                 /mnt/c/Users/yann/Desktop/code/nni/docs/en_US/builtin_tuner.rst: WARNING: document isn't included in any toctree

/mnt/c/Users/yann/Desktop/code/nni/docs/en_US/BohbAdvisor.md:44: WARNING: Pygments lexer name 'yml' is not known
/mnt/c/Users/yann/Desktop/code/nni/docs/en_US/BuiltinTuner.md:68: WARNING: Pygments lexer name 'yml' is not known                                   /mnt/c/Users/yann/Desktop/code/nni/docs/en_US/CommunitySharings/HpoComparision.md:37: WARNING: None:any reference target not found: ../gbdt_example /mnt/c/Users/yann/Desktop/code/nni/docs/en_US/GeneralNasInterfaces.md:114: WARNING: None:any reference target not found: ./multiPhase               /mnt/c/Users/yann/Desktop/code/nni/docs/en_US/GeneralNasInterfaces.md:49: WARNING: Could not lex literal_block as "json". Highlighting skipped.     /mnt/c/Users/yann/Desktop/code/nni/docs/en_US/GeneralNasInterfaces.md:49: WARNING: Could not lex literal_block as "json". Highlighting skipped.     /mnt/c/Users/yann/Desktop/code/nni/docs/en_US/HowToImplementTrainingService.md:10: WARNING: None:any reference target not found: FrameworkController/mnt/c/Users/yann/Desktop/code/nni/docs/en_US/Nnictl.md:741: WARNING: Pygments lexer name 'yml' is not known                                        /mnt/c/Users/yann/Desktop/code/nni/docs/en_US/Overview.md:53: WARNING: None:any reference target not found: BuiltinAssessors                        /mnt/c/Users/yann/Desktop/code/nni/docs/en_US/QuickStart.md:243: WARNING: None:any reference target not found: BuiltinAssessors                     /mnt/c/Users/yann/Desktop/code/nni/docs/en_US/QuickStart.md:24: WARNING: Pygments lexer name 'pseudo' is not known                                  /mnt/c/Users/yann/Desktop/code/nni/docs/en_US/Release.md:71: WARNING: None:any reference target not found: ./WindowsLocalMode                       /mnt/c/Users/yann/Desktop/code/nni/docs/en_US/Release.md:154: WARNING: more than one target found for 'any' cross-reference 'MetisTuner': could be :doc:`Metis tuner` or :py:class:`nni.metis_tuner.metis_tuner.MetisTuner`
/mnt/c/Users/yann/Desktop/code/nni/docs/en_US/Release.md:156: WARNING: more than one target found for 'any' cross-reference 'CurvefittingAssessor': could be :doc:`Curve fitting assessor` or :py:class:`nni.curvefitting_assessor.curvefitting_assessor.CurvefittingAssessor`
/mnt/c/Users/yann/Desktop/code/nni/docs/en_US/Trials.md:47: WARNING: None:any reference target not found: BuiltinAssessors
```

Solutions:
* fix deadlinks 
* change lexer name: `yml` => `yaml`, `json` => 'javascript'

after fix:
```
/mnt/c/Users/yann/Desktop/code/nni/docs/en_US/community_sharings.rst:7: WARNING: toctree contains reference to nonexisting document 'CommunitySharings/NasComparison'
/mnt/c/Users/yann/Desktop/code/nni/docs/en_US/community_sharings.rst:7: WARNING: toctree contains reference to nonexisting document 'CommunitySharings/HpoComparison'
looking for now-outdated files... none found                                                                                                        pickling environment... done                                                                                                                        checking consistency... /mnt/c/Users/yann/Desktop/code/nni/docs/en_US/CommunitySharings/HpoComparision.md: WARNING: document isn't included in any toctree
/mnt/c/Users/yann/Desktop/code/nni/docs/en_US/CommunitySharings/NasComparision.md: WARNING: document isn't included in any toctree                  /mnt/c/Users/yann/Desktop/code/nni/docs/en_US/GPTuner.md: WARNING: document isn't included in any toctree                                           /mnt/c/Users/yann/Desktop/code/nni/docs/en_US/NniOnWindows.md: WARNING: document isn't included in any toctree                                      /mnt/c/Users/yann/Desktop/code/nni/docs/en_US/builtin_assessor.rst: WARNING: document isn't included in any toctree                                 /mnt/c/Users/yann/Desktop/code/nni/docs/en_US/builtin_tuner.rst: WARNING: document isn't included in any toctree                                    done
preparing documents... done                                                                                                                         writing output... [100%] tutorials                                                                                                                  /mnt/c/Users/yann/Desktop/code/nni/docs/en_US/Release.md:154: WARNING: more than one target found for 'any' cross-reference 'MetisTuner': could be :doc:`Metis tuner` or :py:class:`nni.metis_tuner.metis_tuner.MetisTuner`
/mnt/c/Users/yann/Desktop/code/nni/docs/en_US/Release.md:156: WARNING: more than one target found for 'any' cross-reference 'CurvefittingAssessor': could be :doc:`Curve fitting assessor` or :py:class:`nni.curvefitting_assessor.curvefitting_assessor.CurvefittingAssessor`
```

@QuanluZhang how can I fix `No module named 'ConfigSpace'` error?

### CODE REVIEW

File: docs/en_US/GeneralNasInterfaces.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ To illustrate the convenience of the programming interface, we use the interface
 
 After finishing the trial code through the annotation above, users have implicitly specified the search space of neural architectures in the code. Based on the code, NNI will automatically generate a search space file which could be fed into tuning algorithms. This search space file follows the following JSON format.
 
-```json
+```javascript
```
[the remark this answers is not in the document]
Comment (leckie-chn) [reply docs/en_US/GeneralNasInterfaces.md]: To fix `WARNING: Could not lex literal_block as "json". Highlighting skipped. `

**Comment by QuanluZhang** [comment]
@leckie-chn should install the necessary packages through `nnictl package install`

**Comment by QuanluZhang** [comment]
great job, BTW.

**Comment by leckie-chn** [comment]
> @leckie-chn should install the necessary packages through `nnictl package install`

we can only specify dependencies inside `requirements.txt`, for we cannot customize docker image for  `nni.readthedocs.io`

**Pull request merged**